src/testdir/test_filter_cmd.vim

```diff
@@ -74,3 +74,58 @@ func Test_filter_cmd_completion()
   call assert_equal('filter /pat/ print', s:complete_filter_cmd('filter /pat/ pri'))
   call assert_equal('filter #pat# print', s:complete_filter_cmd('filter #pat# pri'))
 endfunc
+
+func Test_filter_commands()
+  let g:test_filter_a = 1
+  let b:test_filter_b = 2
+  let test_filter_c = 3
+
+  " Test filtering :let command
+  let redi = ""
+  redi => redi
+  filter /^test_filter/ let
```

Comment:
Instead of redirecting the command output to a variable using ":redir",
will it be simpler to use the execute() function?

call assert_equal(["test_filter_a\t#1"], split(execute("filter /^test_filter/ let"), "\n"))